basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/HostTestCase.java

```diff
@@ -37,100 +34,108 @@
-    private final String DEFAULT_WEB_MODULE_ATTR = "default-web-module";
-    private final String DISABLE_CONSOLE_REDIRECT_ATTR = "disable-console-redirect";
-
     //values
     private final String[] ALIAS_VALUES = new String[]{"localhost", "test", "example"};
-    private final String DEFAULT_RESPONSE_CODE_VALUE = "500";
+    private final int DEFAULT_RESPONSE_CODE_VALUE = 500;
+
+    private static final String HTTP_SERVER = "undertow-http-server-host_" + RandomStringUtils.randomAlphanumeric(5);
+
+    private static final String HOST = "host_" + RandomStringUtils.randomAlphanumeric(5);
+    private static final String HOST_TBR = "host-btr_" + RandomStringUtils.randomAlphanumeric(5);
+    private static final String HOST_TBA = "host-tba_" + RandomStringUtils.randomAlphanumeric(5);
+
+    private static final Address HTTP_SERVER_ADDRESS = UNDERTOW_ADDRESS.and("server", HTTP_SERVER);
+
+    private static final Address HOST_ADDRESS = HTTP_SERVER_ADDRESS.and("host", HOST);
+    private static final Address HOST_TBR_ADDRESS = HTTP_SERVER_ADDRESS.and("host", HOST_TBR);
+    private static final Address HOST_TBA_ADDRESS = HTTP_SERVER_ADDRESS.and("host", HOST_TBA);
 
-    private static AddressTemplate hostTemplate = httpServerTemplate.append("/host=*");
-    private static String httpServer;
-    private static String httpServerHost;
-    private static String httpServerHostToBeRemoved;
-    private static ResourceAddress address;
 
     @BeforeClass
     public static void setUp() throws InterruptedException, TimeoutException, IOException {
-        httpServer = operations.createHTTPServer();
-        httpServerHost = operations.createHTTPServerHost(httpServer);
-        httpServerHostToBeRemoved = operations.createHTTPServerHost(httpServer);
-        address = hostTemplate.resolve(context, httpServer, httpServerHost);
+        operations.add(HTTP_SERVER_ADDRESS);
+        administration.reloadIfRequired();
+
+        operations.add(HOST_ADDRESS);
+        operations.add(HOST_TBR_ADDRESS);
+        administration.reloadIfRequired();
     }
 
     @Before
     public void before() {
         page.navigate();
-        page.viewHTTPServer(httpServer).switchToHosts().selectItemInTableByText(httpServerHost);
+        page.viewHTTPServer(HTTP_SERVER)
+                .switchToHosts()
+                .selectItemInTableByText(HOST);
     }
 
     @AfterClass
-    public static void tearDown() throws InterruptedException, IOException, TimeoutException {
-        operations.removeHTTPServerHostIfExists(httpServer, httpServerHost);
-        operations.removeHTTPServer(httpServer);
+    public static void tearDown() throws InterruptedException, IOException, TimeoutException, OperationException {
+        operations.removeIfExists(HOST_ADDRESS);
+        operations.removeIfExists(HOST_TBA_ADDRESS);
+        operations.removeIfExists(HOST_TBR_ADDRESS);
+        operations.removeIfExists(HTTP_SERVER_ADDRESS);
+
+        administration.reloadIfRequired();
```

Comment:
Suggesting to switch order of `reloadIfRequired()` and `restertIfRequired()`. See comment above.